Kapacitor Flux tasks (#2687)
influxdata/docs-v2: content/kapacitor/v1.6/working/flux/_index.md

---
title: Use Flux tasks with Kapacitor
description: >
  Use **Kapacitor 1.6+** to run Flux tasks against InfluxDB and other data sources.
  Leverage the full library of Flux functionality to build powerful data processing
  and monitoring tasks in Kapacitor.
menu:
  kapacitor_1_6:
    name: Use Flux tasks
    parent: work-w-kapacitor
related: 
  - /influxdb/cloud/tools/kapacitor/
  - /{{< latest "influxdb" >}}/tools/kapacitor/
  - /kapacitor/v1.6/working/flux/manage/create/
  - /influxdb/cloud/process-data/get-started/, Get started with Flux tasks
  - /influxdb/cloud/process-data/common-tasks/
  - /influxdb/cloud/process-data/task-options/
---

Use **Kapacitor 1.6+** to run Flux tasks against InfluxDB and other data sources.
Leverage the full library of Flux functionality to build powerful data processing
and monitoring tasks in Kapacitor.

## Before you start

> Before you get started, there are things to consider when using Kapacitor Flux tasks:

Reviewer: ```suggestion
Before you get started with Flux tasks, consider:
```

> - Kapacitor Flux tasks do not use Kapacitor topics or event handlers.

Reviewer: Do not or _cannot_?

Author: Umm both 😄

    `database-name/retention-policy-name` pattern to specify your bucket.

    ##### example-task.flux
    {{< keep-url >}}
    ```js
    option task = {
      name: "example-task-name",
      every: 1h,
      offset: 10m
    }

    host = "http://localhost:8086"
    token = ""

    from(bucket: "example-db/example-rp", host: host, token: token)
      |> range(start: -task.every)
      |> filter(fn: (r) => r._measurement == "example-measurement")
      |> aggregateWindow(every: 10m, fn: mean)
      |> to(bucket: "example-db/example-rp-downsampled", host: host, token: token)
    ```

2. Use the `kapacitor flux task create` command to add your Flux script as a Kapacitor Flux task.

    ```sh
    kapacitor flux task create --file /path/to/example-task.flux
    ```

_For more details about creating Kapacitor Flux tasks, see [Create a Kapacitor Flux task](/kapacitor/v1.6/working/flux/manage/create/)._
{{% /tab-content %}}
<!-------------------------- END InfluxDB 1.x content ------------------------->
<!---------------------- BEGIN InfluxDB Cloud/2.x content --------------------->
{{% tab-content %}}
{{% warn %}}
#### Consider using InfluxDB tasks

> If you're using **InfluxDB Cloud** or **InfluxDB OSS 2.x**, you certainly can
> use Kapacitor to run Flux tasks against them, but you should consider just using
> [native InfluxDB tasks](/influxdb/cloud/process-data/).

Reviewer: ```suggestion
If you're using **InfluxDB Cloud** or **InfluxDB OSS 2.x**, consider using
[native InfluxDB tasks](/influxdb/cloud/process-data/) for data processing.
```
